pkg/operator/injection/injection.go

```diff
@@ -68,15 +53,23 @@ func GetControllerName(ctx context.Context) string {
 // dynamically at runtime due to the necessity of having to build logic around re-queueing to ensure that settings are
 // properly reloaded for things like feature gates
 func WithSettingsOrDie(ctx context.Context, kubernetesInterface kubernetes.Interface, settings ...settings.Injectable) context.Context {
-	cancelCtx, cancel := context.WithCancel(ctx)
+	cancelCtx, cancel := context.WithTimeout(ctx, 10*time.Second)
 	defer cancel()
 
 	factory := informers.NewSharedInformerFactoryWithOptions(kubernetesInterface, time.Second*30, informers.WithNamespace(system.Namespace()))
 	informer := factory.Core().V1().ConfigMaps().Informer()
 	factory.Start(cancelCtx.Done())
 
 	for _, setting := range settings {
-		cm := lo.Must(WaitForConfigMap(ctx, setting.ConfigMap(), informer))
+		cm, err := WaitForConfigMap(ctx, setting.ConfigMap(), informer)
+		if err != nil {
```

Comment:
```suggestion
		if client.IgnoreNotFound(err) != nil {
```

Can shorten this and drop the second if

Reply:
Not sure if this is what you had in mind but this is what I think it can be simplified to:
```go
if client.IgnoreNotFound(err) != nil {
    panic(err)
} else if err != nil {
    continue
}
```
It still needs the two if statements but gets rid of the nesting.

Reply:
Oop. My fault, I missed L80 and I see why you can't drop the if check completely. I actually like the old way that you had it better now that I see L80 😛